app/controllers/settings/ext_cert_controller.rb

```diff
@@ -0,0 +1,254 @@
+require "openssl"
+
+# Settings::Ext_Cert allows users to install their own SSL Certificates
+# and Private Keys for encrypted external communication
+# rubocop:disable Metrics/ClassLength
+class Settings::ExtCertController < SettingsController
+  VELUM_NAME = "Velum".freeze
+  KUBEAPI_NAME = "Kubernetes API".freeze
+  DEX_NAME = "Dex".freeze
+
+  def index
+    set_instance_variables
+  end
+
+  def create
+    key_cert_map_temp = key_cert_map
+    key_cert_map_temp.each_key do |i|
+      return false unless upload_validate(key_cert_map_temp[i])
+    end
+
+    cert_map = {
+      external_cert_velum_cert:   key_cert_map_temp[:velum][:cert][:cert_string],
+      external_cert_velum_key:    key_cert_map_temp[:velum][:key][:key_string],
+      external_cert_kubeapi_cert: key_cert_map_temp[:kubeapi][:cert][:cert_string],
+      external_cert_kubeapi_key:  key_cert_map_temp[:kubeapi][:key][:key_string],
+      external_cert_dex_cert:     key_cert_map_temp[:dex][:cert][:cert_string],
+      external_cert_dex_key:      key_cert_map_temp[:dex][:key][:key_string]
+    }
+    @errors = Pillar.apply cert_map
+    if @errors.empty?
+      redirect_to settings_ext_cert_index_path,
+        notice: "External Certificate settings successfully saved."
+      return
+    # :nocov:
+    # An error here would require a failure in connection to velum->salt
+    # or a corruption in mapping of values in the salt pillar
+    else
+      set_instance_variables
+      render action: :index, status: :unprocessable_entity
+      # :nocov:
+    end
+  end
+
+  private
+
+  def set_instance_variables
+    @velum_cert = cert_parse(Pillar.value(pillar: :external_cert_velum_cert))
+    @velum_key = key_parse(Pillar.value(pillar: :external_cert_velum_key))
+    @kubeapi_cert = cert_parse(Pillar.value(pillar: :external_cert_kubeapi_cert))
+    @kubeapi_key = key_parse(Pillar.value(pillar: :external_cert_kubeapi_key))
+    @dex_cert = cert_parse(Pillar.value(pillar: :external_cert_dex_cert))
+    @dex_key = key_parse(Pillar.value(pillar: :external_cert_dex_key))
+  end
+
+  def get_val_from_form(form, param)
+    if params[form][param].present?
```

Comment:
Isn't better if we hard code `params[:external_certificate]` since it will always be the same param prefix? it will make the call shorter. Or do we plan to pass other params outside `:external_certificate`?

Reply:
Also, if the user submits the form without selecting any file, a nil exception occurs.

Reply:
Hard-coding `:external_certificate` is fine with me.

Reply:
Nil exception fixed